gix-pack/src/data/file/decode/entry.rs

```diff
@@ -400,7 +399,8 @@ impl File {
             // this seems inverted, but remember: we swapped the buffers on the last iteration
             target_buf[..last_result_size].copy_from_slice(&source_buf[..last_result_size]);
         }
-        out.resize(last_result_size, 0);
+        debug_assert!(out.len() >= last_result_size);
```

Comment:
Thank you, this assertion expresses the intend much better than a blank `resize` which should always truncate the vec anyway.